### PR TITLE
Feature/mp 1149 custom checkout customer details

### PR DESCRIPTION
**Pull request by danial-rokt**
Simulator Screen Shot - iPhone 12 - 2021-07-09 at 13 25 25

### CODE REVIEW

File: RoktDemo/Service/DemoLibraryService.swift [hunk 1]
```diff
@@ -64,11 +64,11 @@ This allows you to preview the specific UI of this placement and experience the
...
 """,
                                                              iconURL: "CustomerCheckout",
-                                                             customerDetails:
-                                                                CustomerDetailsModel(accountID: "2754655826098840951",
+                                                             accountDetails:
+                                                                AccountDetailsModel(accountID: "2754655826098840951",
                                                                                      viewName: "tesLightboxiOS",
                                                                                      placementLocation1: "",
-                                                                                     placementLocation2: "")),
+                                                                                     placementLocation2: ""),customerDetails: CustomerDetailsModel(state: "NSW", postcode: "2000", country: ["US","AU","UK","SG"]), advancedDetails: ["experience":"true","majorCat":"true"]),
```
Comment (DJingSeo) [inline RoktDemo/Service/DemoLibraryService.swift:71]: split to next lines? this line is too long

File: RoktDemo/ViewModels/CustomerDetailsViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        state = customerDetails.state
+        postcode = customerDetails.postcode
+        countries = customerDetails.country
+        selectedCountry = countries.first ?? ""
+        
+        advancedDetailsKV = []
+        for (key, value) in advancedDetails {
+            advancedDetailsKV.append(KeyValue(key: key, value: value))
+        }
+        // add empty key value at the end
```
Comment (DJingSeo) [inline RoktDemo/ViewModels/CustomerDetailsViewModel.swift:38]: More comments please. I dont know why we need an empy key value pair.

File: RoktDemo/Views/AccountDetailsView.swift [hunk 3]
```diff
@@ -73,10 +80,10 @@ struct AccountDetailsView_Previews: PreviewProvider {
...
                                                          longDescription: "",
                                                          iconURL: "",
-                                                         customerDetails:
-                                                            CustomerDetailsModel(accountID: "123",
+                                                         accountDetails:
+                                                            AccountDetailsModel(accountID: "123",
                                                                                  viewName: "",
                                                                                  placementLocation1: "",
-                                                                                 placementLocation2: ""))), popToRootView: .constant(true))
+                                                                                 placementLocation2: "") ,customerDetails: CustomerDetailsModel(state: "", postcode: "", country: []), advancedDetails: [:])), popToRootView: .constant(true))
```
Comment (DJingSeo) [inline RoktDemo/Views/AccountDetailsView.swift:87]: Same as above. This line is too long

File: RoktDemo/Views/DemoItemSummaryView.swift [hunk 4]
```diff
@@ -106,8 +122,8 @@ struct CustomCheckoutView_Previews: PreviewProvider {
 
    """,
                                                                  iconURL: "CustomerCheckout",
-                                                                 customerDetails:
-                                                                    CustomerDetailsModel(accountID: "", viewName: "", placementLocation1: "", placementLocation2: "")))
+                                                                 accountDetails:
+                                                                    AccountDetailsModel(accountID: "", viewName: "", placementLocation1: "", placementLocation2: ""),customerDetails: CustomerDetailsModel(state: "NSW", postcode: "200", country: ["US","AU","UK","SG"]), advancedDetails: ["experience":"true","majorCat":"true"]))
```
Comment (DJingSeo) [inline RoktDemo/Views/DemoItemSummaryView.swift:126]: Same as above, too long

**Review by DJingSeo — CHANGES_REQUESTED** [review]

File: RoktDemo/Views/DemoItemSummaryView.swift (hunk 4 above)
Comment (danial-rokt) [reply RoktDemo/Views/DemoItemSummaryView.swift]: Done

File: RoktDemo/Views/AccountDetailsView.swift (hunk 3 above)
Comment (danial-rokt) [reply RoktDemo/Views/AccountDetailsView.swift]: Done

File: RoktDemo/ViewModels/CustomerDetailsViewModel.swift (hunk 2 above)
Comment (danial-rokt) [reply RoktDemo/ViewModels/CustomerDetailsViewModel.swift]: Done

File: RoktDemo/Service/DemoLibraryService.swift (hunk 1 above)
Comment (danial-rokt) [reply RoktDemo/Service/DemoLibraryService.swift]: Done

**Review by DJingSeo — APPROVED** [review]

**Pull request merged**